operators/src/util/raster_stream_to_geotiff.rs

```diff
@@ -798,11 +944,88 @@ mod tests {
         //     "../test_data/raster/cloud_optimized_geotiff_from_stream_compressed.tiff",
         // );
 
+        assert_eq!(bytes.len(), 1);
+
         assert_eq!(
             include_bytes!(
                 "../../../test_data/raster/cloud_optimized_geotiff_from_stream_compressed.tiff"
             ) as &[u8],
-            bytes.as_slice()
+            bytes.pop().expect("bytes should have length 1").as_slice()
+        );
+
+        // TODO: check programmatically that intermediate file is gone
+    }
+
+    #[tokio::test]
+    async fn cloud_optimized_geotiff_multiple_timesteps_from_stream() {
+        let ctx = MockQueryContext::test_default();
+        let tiling_specification =
+            TilingSpecification::new(Coordinate2D::default(), [600, 600].into());
+
+        let metadata = create_ndvi_meta_data();
+
+        let gdal_source = GdalSourceProcessor::<u8> {
+            tiling_specification,
+            meta_data: Box::new(metadata),
+            _phantom_data: PhantomData,
+        };
+
+        let query_bbox = SpatialPartition2D::new((-10., 80.).into(), (50., 20.).into()).unwrap();
+
+        let mut bytes = raster_stream_to_geotiff_bytes(
+            gdal_source.boxed(),
+            RasterQueryRectangle {
+                spatial_bounds: query_bbox,
+                time_interval: TimeInterval::new(
+                    1_388_534_400_000,
+                    1_388_534_400_000 + 7_776_000_000,
+                )
+                .unwrap(),
+                spatial_resolution: SpatialResolution::new_unchecked(
+                    query_bbox.size_x() / 600.,
+                    query_bbox.size_y() / 600.,
+                ),
+            },
+            ctx,
+            GdalGeoTiffDatasetMetadata {
+                no_data_value: Some(0.),
+                spatial_reference: SpatialReference::epsg_4326(),
+            },
+            GdalGeoTiffOptions {
+                as_cog: true,
+                compression_num_threads: GdalCompressionNumThreads::AllCpus,
+                force_big_tiff: false,
+            },
+            None,
+            Box::pin(futures::future::pending()),
+        )
+        .await
+        .unwrap();
+
+        assert_eq!(bytes.len(), 3);
+
+        // for i in 0..3 {
+        //     let file_name = format!(
+        //         "../test_data/raster/cloud_optimized_geotiff_timestep_{}_from_stream_compressed.tiff",
+        //         i
+        //     );
+        //     geoengine_datatypes::util::test::save_test_bytes(
+        //         &bytes.pop().expect("bytes should have length 3"),
+        //         file_name.as_str(),
+        //     );
+        // }
```

Comment:
```suggestion
```

Reply:
I believe in other tests those kinds of lines are kept (e.g., cloud_optimized_geotiff_from_stream), because you might need to regenerate those files if the gdal version changes.